pkg/dockerregistry/server/client/test.go

```diff
@@ -3,29 +3,26 @@ package client
 import (
 	kcoreclient "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/core/internalversion"
 
-	"github.com/openshift/origin/pkg/client"
 	imageclientv1 "github.com/openshift/origin/pkg/image/generated/clientset/typed/image/v1"
 )
 
 type fakeRegistryClient struct {
 	RegistryClient
 
-	client client.Interface
 	images imageclientv1.ImageV1Interface
 }
 
-func NewFakeRegistryClient(c client.Interface, imageclient imageclientv1.ImageV1Interface) RegistryClient {
+func NewFakeRegistryClient(imageclient imageclientv1.ImageV1Interface) RegistryClient {
 	return &fakeRegistryClient{
 		RegistryClient: &registryClient{},
-		client:         c,
 		images:         imageclient,
 	}
 }
 
 func (c *fakeRegistryClient) Client() (Interface, error) {
-	return newAPIClient(c.client, nil, nil, c.images, nil), nil
+	return newAPIClient(nil, nil, c.images, nil), nil
```

Comment:
@dmage as part of the test client refactoring we should get rid of all this nils and just pass the clients we actually use (I believe we use just images client).

Reply:
We just don't have unit tests for the paths with other clients (e.g., limit ranges). If we have better code coverage, we will not have nils there.